File: leetsync/3455-minimum-length-of-string-after-operations/minimum-length-of-string-after-operations.go
func minimumLength(s string) int {
	freq := make([]int, 26)
	n := len(s)
	deletedChars := 0
	for i := 0; i < n; i++ {
		chInt := s[i] - 'a'
		freq[chInt]++
		if freq[chInt] == 3 {
			freq[chInt] = 1
			deletedChars += 2
		}
	}
	return n - deletedChars
}

File: leetsync/3455-minimum-length-of-string-after-operations/minimum-length-of-string-after-operations.java
class Solution {
    public int minimumLength(String s) {
        int[] freq = new int[26];
        int n = s.length();
        int deletedChars = 0;
        for (int i = 0; i < n; i++) {
            char ch = s.charAt(i);
            int chInt = ch - 'a';
            freq[chInt]++;
        }
        int length = 0;
        for (int i = 0; i < 26; i++) {
            if (freq[i] == 0) {
                continue;
            }
            if (freq[i] % 2 == 1) {
                length += 1;
            } else {
                length += 2;
            }
        }
        return length;
    }
}